src/core_plugins/tile_map/public/base_maps_visualization.js

```diff
@@ -167,12 +170,10 @@ export function BaseMapsVisualizationProvider(serviceSettings) {
     }
 
     async _setTmsLayer(tmsLayer) {
-      if (tmsLayer.maxZoom < this._kibanaMap.getMaxZoomLevel()) {
```

Comment:
this was the bug. This check only worked in the wrong direction. ie. being in a current config with more zoom levels, and having to snap back because the new config has fewer zoom levels. But it should work the other way around as well. If you have more zoom levels, it should enable these too.